galasa-ui/src/tests/components/ArtifactsTab.test.tsx

```diff
@@ -0,0 +1,457 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor, act } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { ArtifactsTab } from '@/components/test-runs/ArtifactsTab';
+import { downloadArtifactFromServer } from '@/actions/runsAction';
+import { handleDownload } from '@/utils/artifacts';
+
+// Mock dependencies
+jest.mock('@/actions/runsAction');
+jest.mock('@/utils/artifacts');
```

Comment:
Worried that we're mocking out behaviour that needs testing, like the logic to download artifacts. Do we need to mock the functions in `artifacts` and `runsAction` out? Are they unit tested elsewhere?